sim/dex-data.ts

```diff
@@ -905,70 +911,72 @@ export class Move extends BasicEffect implements Readonly<BasicEffect & MoveData
 		this.stab = data.stab || undefined;
 		this.volatileStatus = typeof data.volatileStatus === 'string' ? (data.volatileStatus as ID) : undefined;
 
-		if (this.category !== 'Status' && !this.gmaxPower) {
+		if (this.category !== 'Status' && !this.maxMove?.basePower) {
```

Comment:
```suggestion
		if (this.category !== 'Status' && !this.maxMove && this.id !== 'struggle') {
```

The only difference between `!this.maxMove` and `!this.maxMove?.basePower` is that the latter will be truthy if `maxMove.basePower` is `0`. This is undesirable; if a move explicitly specifies a `basePower` of `0` for whatever reason, we should not overwrite it.

Reply:
(If `maxMove.basePower` should never be `0`, that should be enforced in `test/sim/data`, not here.)